File: Data Structure/Queue/implementation/C++/queue_implement_by_array.cpp
#include<iostream>
using namespace std;

int front = 0, rear = 0; 

void enqueue(int queue[], int element, int arrSize)
{
    if(rear != arrSize){
        queue[rear] = element;
        rear++;
    }
}

void dequeue(int queue[])
{
    if(front != rear){
        queue[front] = 0;
        front++;
    }
}

int frontData(int queue[])
{
    return queue[front];
}

int size()
{
    return rear - front;
}

bool isEmpty()
{
    return front == rear;
}

int main()
{
    int n, k;
    cin >> n;
    int queue[n+1];
    for(int i = 0; i < n; i++){
        cin >> k;
        enqueue(queue, k, n);
    }
    cout << "Queue size : " << size() << endl;
    cout << "Queue first element = " << frontData(queue) << endl;
    dequeue(queue);
    cout << "Queue new first element = " << frontData(queue) << endl;
    cout << "Queue is empty = " << (isEmpty()? "True":"False") << endl;
    cout << "Queue data are : " ;
    while(!isEmpty()) {
        cout << frontData(queue) << " ";
        dequeue(queue);
    }
    cout << endl;
    cout << "Queue is empty = " << (isEmpty()? "True":"False") << endl;
    return 0;
}

File: Data Structure/Queue/implementation/python/queue_implement_by_array.py
class Queue:
    def __init__(self):
        self.items = []
    
    def isEmpty(self):
        return self.items == []

    def enqueue(self, item):
        self.items.insert(0,item)
    
    def dequeue(self):
        return self.items.pop()
    
    def size(self):
        return len(self.items)

q = Queue()

q.enqueue(4)
q.enqueue('dogs')
q.enqueue(True)

print("Queue size: ", q.size())
print("Empty or not : ", q.isEmpty())
